core/storage/user/gae_models.py

```diff
@@ -130,7 +130,10 @@ def export_data(user_id):
         Returns:
             dict. Dictionary of the data from UserSettingsModel.
         """
-        user = UserSettingsModel.get(user_id)
+        try:
+            user = UserSettingsModel.get(user_id)
+        except UserSettingsModel.EntityNotFoundError:
+            return {}
```

Comment:
This one should probably throw an exception in this case -- we would expect a UserSettingsModel to exist for each user, so its absence would be unexpected and we should get some sort of alert.

Reply:
This is the case of the user not existing, so the reason why we catch this error is because from an export data perspective, we just want to indicate to the caller that there exists no data for the user requested (because they do not exist). So I think that this is the correct flow, but feel free to correct me.

Reply:
Callers should only be able to request their own data, so I think it's fine to throw an error here.

Reply:
OK so then do I change the behavior everywhere to throw an exception? The most consistent default behavior I thought I could implement was to simply return empty, but if we decide we want to throw an exception in this case, then I would think for consistency sake we would want to throw an exception everywhere.

Reply:
No, only in this case. The more general rule is: if it's possible/reasonable for no data to exist in a particular model for a user, then empty dict (or equivalent) is appropriate. But if we would expect, during normal operation, all users to be represented in the given model, then an exception should be raised if this condition doesn't hold, since it's indicative of a bug with the system.

Reply:
OK so then in the case of the larger takeout_service, when I write a test to export data with a nonexistent user, will I just have my test assert that an exception is thrown? I'm a bit confused by how this decision would play out in the larger context.

Reply:
Exceptions are thrown in the case where we don't expect that part of the code to execute. So, if it does execute, the user should see an error (i.e. the exception should bubble up into a 500 error).